### PR TITLE
changed to signature based file validation (gog installer)

### PR DESCRIPTION
**Pull request by Laserlicht**
use signature based validation instead of file ending

### CODE REVIEW

File: launcher/firstLaunch/firstlaunch_moc.cpp [hunk 1]
```diff
@@ -332,10 +340,10 @@ void FirstLaunchView::extractGogData()
 		return file;
 	};
 
-	QString fileBin = fileSelection("bin", tr("GOG data") + " (*.bin)");
+	QString fileBin = fileSelection("idska32", tr("GOG data") + " (*.bin)");
 	if(fileBin.isEmpty())
 		return;
-	QString fileExe = fileSelection("exe", tr("GOG installer") + " (*.exe)", QFileInfo(fileBin).absolutePath());
+	QString fileExe = fileSelection("MZ", tr("GOG installer") + " (*.exe)", QFileInfo(fileBin).absolutePath());
```
Comment (kambala-decapitator) [inline launcher/firstLaunch/firstlaunch_moc.cpp:346]: technically speaking, we should also check for PE format: https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#signature-image-only

although I think that even this solution is already good

File: launcher/firstLaunch/firstlaunch_moc.cpp [hunk 2]
```diff
@@ -323,7 +323,15 @@ void FirstLaunchView::extractGogData()
...
-		else if(!file.endsWith("." + type, Qt::CaseInsensitive))
+		
+		QFile tmpFile(file);
+		if(!tmpFile.open(QIODevice::ReadOnly))
+		{
+			QMessageBox::critical(this, tr("File cannot opened"), titleErr);
+			return QString{};
+		}
+		QByteArray magicFile = tmpFile.read(magic.length());
+		if(!magicFile.startsWith(QByteArray{magic.toStdString().c_str()}))
```
Comment (kambala-decapitator) [inline launcher/firstLaunch/firstlaunch_moc.cpp:334]: better pass `magic` to the lambda as QByteArray

Comment (Laserlicht) [reply launcher/firstLaunch/firstlaunch_moc.cpp]: Thanks!

File: launcher/firstLaunch/firstlaunch_moc.cpp (hunk 1 above)
Comment (Laserlicht) [reply launcher/firstLaunch/firstlaunch_moc.cpp]: Thanks. Yes, was aware of this.
But I think in our case it's good enough like it is. To keep it simple.

**Review by kambala-decapitator — APPROVED** [review]

**Pull request merged**